### PR TITLE
- Added limit of 100 to the environment name.

### PR DESCRIPTION
**Pull request by tamtamirr**
Limiting environment name to 100 characters.
---

In addition to the new limit, fixed a bug where there were no validations on the environment name during rename action.

---
_Release Notes_: 
Workspace:
- Limiting env names to 100 characters.
- Added validation on env names during rename action.

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/workspace/src/workspace/errors.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import { SaltoError } from '@salto-io/adapter-api'
 import { MergeError } from '../merger'
 import { ValidationError } from '../validator'
 import { ParseError } from '../parser'
+import {MAX_ENV_NAME_LEN} from "./workspace";
```
Comment (amitgoldarb) [inline packages/workspace/src/workspace/errors.ts:22]: We use single quotes as a convention
```suggestion
import {MAX_ENV_NAME_LEN} from './workspace';
```

File: packages/workspace/test/workspace/workspace.test.ts [hunk 2]
```diff
@@ -2128,10 +2130,19 @@ describe('workspace', () => {
       await expect(workspace.addEnvironment(envName, mockRmcToSource))
         .rejects.toEqual(new EnvDuplicationError(envName))
     })
-    it('should throw InvalidEnvNameError', async () => {
-      const invalidEnvName = 'invalid:env'
-      await expect(workspace.addEnvironment(invalidEnvName, mockRmcToSource))
-        .rejects.toEqual(new InvalidEnvNameError(invalidEnvName))
+
+    describe('invalid new environment name', () => {
+
```
Comment (amitgoldarb) [inline packages/workspace/test/workspace/workspace.test.ts:2135]: Unnecessary blank line.

File: packages/workspace/test/workspace/workspace.test.ts [hunk 3]
```diff
@@ -2128,10 +2130,19 @@ describe('workspace', () => {
...
+      test('throws InvalidEnvNameError for name with illegal characters', async () => {
+        const nameWithIllegalChars = 'invalid:env'
+        await expect(workspace.addEnvironment(nameWithIllegalChars, mockRmcToSource))
+            .rejects.toThrow(InvalidEnvNameError)
+      })
+      test('throws InvalidEnvNameError when name is too long', async () => {
+        const longName = new Array(MAX_ENV_NAME_LEN * 2).join('a')
```
Comment (amitgoldarb) [inline packages/workspace/test/workspace/workspace.test.ts:2142]: No need to use Array. There is a native string function that is probably more performant.
```suggestion
        const longName = '#'.repeat(MAX_ENV_NAME_LEN + 1)
```

File: packages/workspace/test/workspace/workspace.test.ts [hunk 4]
```diff
@@ -2386,6 +2397,19 @@ describe('workspace', () => {
...
         })
       })
+      describe('invalid new environment name', () => {
+        test('throws InvalidEnvNameError for name with illegal characters', async () => {
+          const nameWithIllegalChars = 'invalid:env'
+          await expect(workspace.renameEnvironment(workspace.currentEnv(), nameWithIllegalChars))
+              .rejects.toThrow(InvalidEnvNameError)
+        })
+        test('throws InvalidEnvNameError when name is too long', async () => {
+          const longName = new Array(MAX_ENV_NAME_LEN * 2).join('a')
```
Comment (amitgoldarb) [inline packages/workspace/test/workspace/workspace.test.ts:2407]: Same thing here with the long name creation.

File: packages/workspace/src/workspace/errors.ts (hunk 1 above)
Comment (amitgoldarb) [reply packages/workspace/src/workspace/errors.ts:22]: Also this introduces a circular dependency. May I suggest passing the const to the Error in the constructor?

**Review by amitgoldarb — COMMENTED** [review]
Looks great!
Added some comments

File: packages/workspace/src/workspace/errors.ts (hunk 1 above)
Comment (tamtamirr) [reply packages/workspace/src/workspace/errors.ts:22]: V, moved the const to the errors file

File: packages/workspace/test/workspace/workspace.test.ts (hunk 2 above)
Comment (tamtamirr) [reply packages/workspace/test/workspace/workspace.test.ts:2135]: V

File: packages/workspace/test/workspace/workspace.test.ts (hunk 3 above)
Comment (tamtamirr) [reply packages/workspace/test/workspace/workspace.test.ts:2142]: v

File: packages/workspace/test/workspace/workspace.test.ts (hunk 4 above)
Comment (tamtamirr) [reply packages/workspace/test/workspace/workspace.test.ts:2407]: v

**Review by amitgoldarb — APPROVED** [review]
Looks great!
Great work!

**Comment by ori-moisis** [comment]
missing ticket number in the PR title (haven't looked at the code)

**Pull request merged**